openmls/src/group/mls_group/create_commit.rs

```diff
@@ -66,10 +72,41 @@ impl CoreGroup {
 
         let proposal_reference_list = proposal_queue.commit_list();
 
-        let sender_index = self.sender_index();
         // Make a copy of the current tree to apply proposals safely
         let mut diff: TreeSyncDiff = self.treesync().empty_diff()?;
 
+        // If this is not an external commit we have to set our own leaf index
+        // and add our leaf. Also, we have to generate the
+        // [`KeyPackageBundlePayload`] slightly differently, because we can't
+        // just pull it from the tree if we're a `NewMember`.
+        let key_package_bundle_payload = if params.commit_type() == CommitType::External {
+            // Set our own index in the diff.
+            diff.set_own_index(own_leaf_index);
+
+            // Generate a KeyPackageBundle to generate a payload from for later
+            // path generation.
+            let key_package_bundle = KeyPackageBundle::new(
+                &[ciphersuite.name()],
+                params.credential_bundle(),
+                backend,
+                vec![],
+            )?;
+
+            let _leaf_index = diff.add_leaf(key_package_bundle.key_package().clone())?;
+            debug_assert_eq!(own_leaf_index, _leaf_index);
+            KeyPackageBundlePayload::from_rekeyed_key_package(
```

Comment:
You could do this outside of the conditional then you don't have to write it twice.